crates/jit/src/instantiate.rs

```diff
@@ -22,6 +22,30 @@ use wasmtime_environ::{
 use wasmtime_profiling::ProfilingAgent;
 use wasmtime_runtime::{GdbJitImageRegistration, InstantiationError, VMFunctionBody, VMTrampoline};
 
+/// This is the name of the section in the final ELF image which contains
+/// concatenated data segments from the original wasm module.
+///
+/// This section is simply a list of bytes and ranges into this section are
+/// stored within a `Module` for each data segment. Memory initialization and
+/// passive segment management all index data directly located in this section.
+///
+/// Note that this implementation does not afford any method of leveraging the
+/// `data.drop` instruction to actually release the data back to the OS. The
+/// data section is simply always present in the ELF image. If we wanted to
+/// release the data it's probably best to figure out what the best
+/// implementation is for it at the time given a particular set of constraints.
+const ELF_WASM_DATA: &'static str = ".data.wasm";
```

Comment:
```suggestion
const ELF_WASM_DATA: &'static str = ".rodata.wasm";
```
This is a non-writeable section.